### DISCUSSION TITLE
Show a custom shape to represent data points

### DISCUSSION THREAD

**Discussion by Praveen-Android** [patrykandpatrick/vico#874] [Questions] [answered]
### Question

Is is possible to show a custom shape(circle etc.) to represent each data point on the chart? Something like this

![image](https://github.com/user-attachments/assets/df7a5e4a-da93-4fbc-a661-4da37cf4692e)

### Vico version(s)

2.0.0-alpha.28

### UI framework(s)

Jetpack Compose

**Comment by Praveen-Android** [comment, marked as answer]
Answering my question here in case anyone is interested:
Implemented this using `LineCartesianLayer.PointProvider ` on a `LineCartesianLayer.Line ` using `LayeredComponent` (since i needed a circle with a hole and padding) but could use `ShapeComponent` for other needs. Its very customizable which is great!

**Reply by patrickmichalik** [reply]
Thanks for sharing! Please note that `LayeredComponent` isn’t required here—`ShapeComponent` lets you add both a fill and a stroke, making it sufficient for creating such an outlined circle.
